# My pipeline for scraping the data & using thecypher package:

In [1]:
import thecypher as cy

In [2]:
logic = cy.get_lyrics('logic')

/anaconda3/lib/python2.7/site-packages/bs4/__init__.py:181: UserWarning: No parser was explicitly specified, so I'm using the best available HTML parser for this system ("lxml"). This usually isn't a problem, but if you run this code on another system, or in a different virtual environment, it may use a different parser and behave differently.

The code that caused this warning is on line 162 of the file /anaconda3/lib/python2.7/runpy.py. To get rid of this warning, change code that looks like this:

 BeautifulSoup(YOUR_MARKUP})

to this:

 BeautifulSoup(YOUR_MARKUP, "lxml")

  markup_type=markup_type))
2018-06-03 09:51:58,152 : INFO : GET Artist URL: http://lyrics.wikia.com/wiki/Logic
2018-06-03 09:51:58,685 : INFO : GET Artist Album: Welcome to Forever (2013)
2018-06-03 09:52:06,443 : INFO : GET Artist Album Successful: Welcome to Forever (2013)
2018-06-03 09:52:06,445 : INFO : GET Artist Album: Misc (0000)
2018-06-03 09:52:06,447 : INFO : GET Artist Album Successful: Misc (0000)
201

In [3]:

import pandas as pd
pd.set_option('display.max_colwidth', -1)



logic_df = pd.DataFrame(logic)
logic_df.head()

album album_genre artist    genre  id  \
0  Welcome to Forever (2013)              logic  Hip Hop  0    
1  Welcome to Forever (2013)              logic  Hip Hop  1    
2  Welcome to Forever (2013)              logic  Hip Hop  2    
3  Welcome to Forever (2013)              logic  Hip Hop  3    
4  Welcome to Forever (2013)              logic  Hip Hop  4    

                                                                 lyric song  \
0  I've seen a lot of shit I shouldn't have but never forgot it though  5AM   
1  Brothers on the corners selling crack like it was nada though        5AM   
2  Walk inside my kitchen baking soda all up on the floor               5AM   
3  Police banging on the door                                           5AM   
4  While gripping a.44                                                  5AM   

   year  
0  2013  
1  2013  
2  2013  
3  2013  
4  2013

In [4]:
logic_df.album.value_counts()

Everybody (2017)                    1422
Young Sinatra (2011)                1259
Under Pressure (2014)               1114
The Incredible True Story (2015)    1075
Undeniable (2012)                   785 
Bobby Tarantino (2016)              676 
Edit                                116 
Welcome to Forever (2013)           65  
Name: album, dtype: int64

#LOGIC 
- Everybody (1422)
- The Incredible True Story (1075)




#BT 
- BT1 (676)




#YOUNG SINATRA 
- Young Sinatra (1259) 
- Undeniable (785)

In [5]:
logic_df = logic_df[logic_df.album != 'Under Pressure (2014)']

In [6]:
logic_df = logic_df[logic_df.album != 'Edit']

In [7]:
logic_df = logic_df[logic_df.album != 'Welcome to Forever (2013)']

In [8]:
logic_df.album.value_counts()

Everybody (2017)                    1422
Young Sinatra (2011)                1259
The Incredible True Story (2015)    1075
Undeniable (2012)                   785 
Bobby Tarantino (2016)              676 
Name: album, dtype: int64

In [9]:
logic_df.reset_index(inplace=True)

In [10]:
logic_df.drop(['index', 'album_genre', 'genre', 'id', 'year'], axis=1, inplace=True)

In [11]:
logic_df.head()

album artist  \
0  Undeniable (2012)  logic   
1  Undeniable (2012)  logic   
2  Undeniable (2012)  logic   
3  Undeniable (2012)  logic   
4  Undeniable (2012)  logic   

                                                            lyric        song  
0  Yeah; pass the mic before I jack it like gore-tex               Disgusting  
1  Bust like raw sex, rappers suck like vortex                     Disgusting  
2  The life of a Don- We living like kings, and killing our pawns  Disgusting  
3  Boy, the seconds it's on - don't know where we going            Disgusting  
4  I'm flowing and killing this shit from dusk till dawn           Disgusting

In [12]:
logic_df.loc[logic_df['album'] == 'Young Sinatra (2011)', "artist"] = "Young Sinatra"
logic_df.loc[logic_df['album'] == 'Undeniable (2012)', "artist"] = "Young Sinatra"
logic_df.loc[logic_df['album'] == 'Bobby Tarantino (2016)', "artist"] = "bobby tarantino"

In [13]:
logic_df.artist.value_counts()

logic              2497
Young Sinatra      2044
bobby tarantino    676 
Name: artist, dtype: int64

In [14]:
logic_df.to_csv('logic_df_part1.csv', encoding='utf-8')